### DISCUSSION TITLE
Change The Color Of The Label When Changing The Component's Status

### DISCUSSION THREAD

**Discussion by QuangLanh** [clientIO/joint#2418] [Q&A]
### Introduction

Currently, I want to add a status selection dropdown for a components so that users can differentiate between components that are in progress and those that have completed their tasks.

If I select the 'In Progress' status, the label of that component should turn red. Conversely, if I choose the 'Completed' status, the label of that component should turn blue.

Is it possible to achieve something like the example shown below?
Please provide suggestions. 

Thank you very much.

### Steps to reproduce

![image](https://github.com/clientIO/joint/assets/91791283/b0ce4635-74c8-4971-8db4-9a2a0c9d6553)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by Geliogabalus** [comment]
Hi! We prepared a [small demo](https://codepen.io/jointjs/pen/ExrrvdN/a80d542d33804801a171106704ad7a72) which demonstrates this functionality.

If you have more questions feel free to ask them here.

**Comment by QuangLanh** [comment]
Your solution needs creating new class. But I cannot use that. Please give solution that works with available class. 
In file inspector-service.ts I added 1 select box , but I don't know how to handle event on change in this select box to change color of label when status changes.

**I provide my code:**
` 'standard.Image': {
        inputs: {
          attrs: {
            label: {
              text: {
                type: 'content-editable',
                label: i18n.t('createDiagram.text'),
                group: 'text',
                index: 1
              },
              fontSize: {
                type: 'range',
                min: 5,
                max: 80,
                unit: 'px',
                label: i18n.t('createDiagram.fontSize'),
                group: 'text',
                when: { ne: { 'attrs/label/text': '' } },
                index: 2
              },
              fontFamily: {
                type: 'select-box',
                options: options.fontFamily,
                label: i18n.t('createDiagram.fontFamily'),
                group: 'text',
                when: { ne: { 'attrs/label/text': '' } },
                index: 3
              },
              fontWeight: {
                type: 'select-box',
                options: options.fontWeight,
                label: i18n.t('createDiagram.fontThickness'),
                group: 'text',
                when: { ne: { 'attrs/label/text': '' } },
                index: 4
              },
              ///PLEASE FIX THIS
              DynamicLabelElement: {
                attrs: {},
                status: {
                  label: 'Status',
                  type: 'select-box',
                  when: { ne: { 'attrs/label/DynamicLabelElement/status': '' } },
                  width: '220',
                  options: [
                    {
                      content: 'Default',
                      value: 'default'
                    },
                    {
                      content: 'In Progress',
                      value: 'inProgress'
                    },
                    {
                      content: 'Completed',
                      value: 'completed'
                    }
                  ]
                }
              }

              // fill: {
              //   type: 'color-palette',
              //   options: options.colorPalette,
              //   label: i18n.t('createDiagram.fill'),
              //   group: 'text',
              //   when: { ne: { 'attrs/label/text': '' } },
              //   index: 5
              // }
            },
            image: {
              xlinkHref: {
                type: 'select-box',
                options: options.imageIcons,
                label: i18n.t('createDiagram.img'),
                group: 'presentation',
                index: 1
              }
            }
          }
        },
        groups: {
          presentation: {
            label: i18n.t('createDiagram.presentation'),
            index: 1
          },
          text: {
            label: i18n.t('createDiagram.text'),
            index: 2
          }
        }
      }`

      
      

![image](https://github.com/clientIO/joint/assets/91791283/4be70c13-24dd-480d-819c-810ee2222fca)

**Reply by kumilingus** [reply]
Why is it not possible to define a custom shape? You can redefine the `standard.Image` with custom shape if needed.

Note, that you can also listen to the changes on the graph level.
```js
graph.on('change:status', (element, status, opt) {
  element.attr("label/fill", StatusColors[status], opt);
});
```

**Comment by QuangLanh** [comment]
From what I see, the code snippet you provided seems different from the code I purchased from you. So, I'm not sure where to copy and paste them. 

Currently, in the "inspector-service-ts" file, I have pasted a portion of your example code, and it has appeared on the interface. 

However, I'm unable to change the status or color of the label as required, and I'm encountering some strange errors. Therefore, I don't know how to handle them.

Here are my four code files:

[app-shapes-ts.txt](https://github.com/clientIO/joint/files/13669032/app-shapes-ts.txt)
[inspector-service-ts.txt](https://github.com/clientIO/joint/files/13668951/inspector-service-ts.txt)
[kitchensink-sevice-ts.txt](https://github.com/clientIO/joint/files/13668952/kitchensink-sevice-ts.txt)
[stencil-service-ts.txt](https://github.com/clientIO/joint/files/13668953/stencil-service-ts.txt)

I hope you can review them and make direct edits to my code snippets.

**Reply by kumilingus** [reply]
As part of our free support, we can only provide you with advice and general examples.
If you need us to implement functionality specifically for your app, you should consider subscribing to one of our [support packages](https://www.jointjs.com/support).

What are the errors that you experience?

**Reply by QuangLanh** [reply]
i can change color label, i have another question
how to change label positon "DS2" (top, left, right, bottom) default postion right, 
![image](https://github.com/clientIO/joint/assets/91791283/99623510-8fb0-4b7b-8eda-59dc383b01aa)
this is problem
![image](https://github.com/clientIO/joint/assets/91791283/1a9647e6-cb90-46c8-8da8-a634062effba)

**Reply by kumilingus** [reply]
These 2 examples should help you with the implementation.
https://www.jointjs.com/demos/change-element-text-alignment-in-inspector
https://www.jointjs.com/demos/text-position-based-on-space-availability

**Reply by QuangLanh** [reply]
ok, thank you so much

**Comment by QuangLanh** [comment]
How can I create 1 connection from one component to another connection and newly created connectionstill straight?
It's like when two components connect together.


![image](https://github.com/clientIO/joint/assets/91791283/46c0413a-dfcd-4f4f-ab06-0d16ccd4c1e5)

**Reply by kumilingus** [reply]
Please create a discussion for each new question. Thank you.

**Reply by QuangLanh** [reply]
Sorry for the inconvenience. Because at this discussion, a member of your team asked me: " If you have more questions feel free to ask them here..".
I created new discussion. please recheck and reply me. 
Thank you so much!
